index.js

```diff
@@ -3,231 +3,249 @@ var BluebirdPromise = require('bluebird');
 var fs = require('fs');
 var _ = require('underscore');
 
-var awsS3Promisified = {
+module.exports = function (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) {
 
-  initialize: function(AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) {
-
-    if (AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY) {
-      AWS.config.update({
-        accessKeyId: AWS_ACCESS_KEY_ID,
-        secretAccessKey: AWS_SECRET_ACCESS_KEY
-      });
-    }
-
-    // default expiration time to 8 hours
-    this.setExpirationInSeconds(28800);
+  if (AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY) {
+    AWS.config.update({
+      accessKeyId: AWS_ACCESS_KEY_ID,
+      secretAccessKey: AWS_SECRET_ACCESS_KEY
+    });
+  }
 
-    return this;
-  },
+  // default expiration time to 8 hours
+  setExpirationInSeconds(28800);
 
-  setExpirationInSeconds: function(t) {
+  function setExpirationInSeconds (t) {
     this.expirationTime = t;
-  },
+  };
 
-  getExpirationInSeconds: function() {
-    return this.expirationTime;
-  },
+  return {
```

Comment:
This way you could easily extract the awsS3Promisified object into a separate file if you want.